### ISSUE TITLE
Dart & Flutter Extension not work with webserver

### ISSUE THREAD

**Issue by fcioffi** [Dart-Code/Dart-Code#2547]
Hi guys
When launch a project from VSCode (F5) it open chrome that show a blank page. Clicking on Dart Debug Extension button on Chrome nothing happens.
Thanks for the support.
Francesco

**Comment by DanTup** [comment]
Is there anything listed in the dev console, either for the current page you have open, or for the Dart debug extension (you may need to go into extensions, enabled Developer Mode and click Inspect Views -> Background Page to view any errors from the extension).

Are you running this all locally on your machine? (If so, out of curiosity - why are you using the Web Server and not the Chrome device?)

**Comment by fcioffi** [comment]
Hi in the dev console there aren't errors. In Debug Extension there is this error:
DevTools failed to load SourceMap: Could not load content for chrome-extension://eljbmlghnomdjgdjmbdekegdkbabckhm/background.dart.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME

I'm using webserver because I develop using WSL.

Thanks for the support!

**Comment by DanTup** [comment]
> I'm using webserver because I develop using WSL

Ah got, it. Are you using WSL2 or original WSL? There are some known network issues with the original WSL with Dart (see https://github.com/microsoft/WSL/issues/4031) - I don't know if it's the cause of this though.

If you inspect the background page for the extension and then click the extensions button, are there any network requests on the network tab?

**Comment by fcioffi** [comment]
I'm using wsl 1, now i try to upgrade to wsl2.
In the meantime, in the network I see a pending request to sseHandler.
Thanks for your support, I upgrade WSL and retry.

**Comment by DanTup** [comment]
> In the meantime, in the network I see a pending request to sseHandler.

It's possible that's related to the WSL issue, though it could also be something else. It'd definitely be better to try and use WSL2 though since WSL is known to have issues (however, I'm not sure if WSL2 is available on stable Windows yet - last time I checked it required the Insiders builds).

If you are able to try it and still have issues though, please let me know!

**Comment by ErikGrimes** [comment]
@DanTup I'm experiencing the same issue using a WSL2 remote.  The browser opens the page, but the application never loads. The VS Code extension sits there waiting for a connection from the Dart Debug extension. 


_Application Console_
```
DevTools failed to load SourceMap: Could not load content for chrome-extension://hdokiejnpimakedhajhdlcegeplioahd/sourcemaps/onloadwff.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
```

_Extension Console_
```
DevTools failed to load SourceMap: Could not load content for chrome-extension://eljbmlghnomdjgdjmbdekegdkbabckhm/background.dart.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
```

Also, WSL2 is now generally available.

**Comment by DanTup** [comment]
>  I'm experiencing the same issue using a WSL2 remote. The browser opens the page, but the application never loads. The VS Code extension sits there waiting for a connection from the Dart Debug extension.

Just to be sure - you do have the [Dart Debug extension](https://chrome.google.com/webstore/detail/dart-debug-extension/eljbmlghnomdjgdjmbdekegdkbabckhm) installed, and you're clicking its button in Chrome when this happens?

If so, can you open your Chrome extensions page, enable Dev mode, then hit Inspect on the background page for the Dart Debug extension and see if there are any errors listed in its console. Thanks!

**Comment by ErikGrimes** [comment]
The second error is from the Dart Debug extension console.  I'm not sure the Dart Debug extension is relevant. I installed simply because of the message in the Debug console thinking it was required.

```Waiting for connection from Dart debug extension at http://localhost:51282```

It seems to be more of an issue with Web Server target, which is the only device initially available in WSL2.  If I run flutter natively on Windows using the Web Server target, I experience the same issue with the application not loading.  I tried it with and without the Dart Debug extension just in case.   IWhat is the Web Server target and should it work?

I'm going to see if I can't get flutter to call Chrome using CHROME_EXECUTABLE.

**Comment by DanTup** [comment]
> I'm not sure the Dart Debug extension is relevant. I installed simply because of the message in the Debug console thinking it was required.

The Dart Debug extension is required to use the web-server device. In order to debug in the browser, we need a connection from VS Code back to the browsers debugger. When using the Chrome device, Chrome is started with some flags that allow us to connect to it. However when running in WSL (where the browser is not launched by our process) we need another way to communicate with the debugger - that's that this extension is for. The extension connects back to the server and proxies the debug requests to the backend (which is running inside the WSL instance).

> It seems to be more of an issue with Web Server target, which is the only device initially available in WSL2. 

Yep, that's expected. We can't launch a browser from inside a remote extension host (or at least, generally... it's possible in WSL there are more options, but this code is designed to support SSH, Docker, and GitPod/Codespaces too).

> I'm going to see if I can't get flutter to call Chrome using CHROME_EXECUTABLE.

If you can make it so your WSL instance looks like it has Chrome installed, can be launched with the required flags, and the debugger port is accessible from within the WSL instance, that might work. However it's not really a supported way. Using the Dart Debugger extension is the expected way to debug web apps that aren't running on the local OS.

To use the Dart Debug extension, all you need to do is click its icon in Chrome once the white page loads (there's an issue about making this more obvious at https://github.com/flutter/flutter/issues/48803).

**Comment by ErikGrimes** [comment]
I see.  I figured out what the issue is with WSL2.  The port number of the URL that gets opened in the browser is different than the one that flutter web server is listening on. If I manually correct the port number and click the Dart Debug Extension, the application loads.  It seems to consistently open the browser to the web server port + 1.

**Comment by DanTup** [comment]
> If I manually correct the port number and click the Dart Debug Extension, the application loads. Flutter or the flutter extension seems to consistently open the browser to the web server port + 1.

Could you capture a log (using **Dart: Capture Logs** and leaving everything ticked except Analyzer), starting before you launch the app?

The URL is mapped using an API in VS Code - it definitely shouldn't return the wrong port, that sounds very wrong 😕

**Comment by ErikGrimes** [comment]
Here is the log capture.

```
!! PLEASE REVIEW THIS LOG FOR SENSITIVE INFORMATION BEFORE SHARING !!

Dart Code extension: 3.12.1
Flutter extension: 3.12.1 (activated)

App: Visual Studio Code - Insiders
Remote: wsl
Version: 1.47.0-insider
Platform: linux

Workspace type: Flutter
Multi-root?: false

Dart SDK:
    Loc: /home/egrimes/opt/flutter/bin/cache/dart-sdk
    Ver: 2.9.0-14.1.beta
Flutter SDK:
    Loc: /home/egrimes/opt/flutter
    Ver: 1.19.0-4.3.pre

HTTP_PROXY: undefined
NO_PROXY: undefined

Fri Jul 03 2020 [11:26:55 GMT-0500 (Central Daylight Time)] Log file started
[11:26:58 AM] [General] [Info] Starting debug session...
[11:26:58 AM] [General] [Info]     workspace: /home/egrimes/documents/development/sandbox/demo
[11:26:58 AM] [General] [Info] Using found common entry point: /home/egrimes/documents/development/sandbox/demo/lib/main.dart
[11:26:58 AM] [General] [Info] Using workspace as cwd: /home/egrimes/documents/development/sandbox/demo
[11:26:58 AM] [General] [Info] Detected launch project as Flutter
[11:26:58 AM] [General] [Info] Using Flutter debug adapter for this session
[11:26:58 AM] [FlutterDaemon] [Info] ==> [{"id":"6","method":"daemon.getSupportedPlatforms","params":{"projectRoot":"/home/egrimes/documents/development/sandbox/demo"}}]
[11:26:58 AM] [FlutterDaemon] [Info] <== [{"id":"6","result":{"platforms":["ios","android","web"]}}]
[11:26:58 AM] [General] [Info] Checking for errors before launching
[11:26:58 AM] [General] [Info] Debug session starting...
    {
        "type": "dart",
        "request": "launch",
        "program": "/home/egrimes/documents/development/sandbox/demo/lib/main.dart",
        "cwd": "/home/egrimes/documents/development/sandbox/demo",
        "name": "Flutter (Web Server)",
        "dartVersion": "2.9.0-14.1.beta",
        "toolEnv": {
            "FLUTTER_HOST": "VSCode",
            "PUB_ENVIRONMENT": "vscode.dart-code"
        },
        "sendLogsToClient": true,
        "globalFlutterArgs": [],
        "args": [],
        "vmAdditionalArgs": [],
        "vmServicePort": 0,
        "dartPath": "/home/egrimes/opt/flutter/bin/cache/dart-sdk/bin/dart",
        "maxLogLineLength": 2000,
        "pubPath": "/home/egrimes/opt/flutter/bin/cache/dart-sdk/bin/pub",
        "pubSnapshotPath": "/home/egrimes/opt/flutter/bin/cache/dart-sdk/bin/snapshots/pub.dart.snapshot",
        "debugSdkLibraries": false,
        "debugExternalLibraries": false,
        "showDartDeveloperLogs": true,
        "useFlutterStructuredErrors": true,
        "evaluateGettersInDebugViews": true,
        "evaluateToStringInDebugViews": true,
        "flutterVersion": "1.19.0-4.3.pre",
        "forceFlutterVerboseMode": true,
        "flutterTrackWidgetCreation": true,
        "flutterMode": "debug",
        "flutterPlatform": "default",
        "flutterPath": "/home/egrimes/opt/flutter/bin/flutter",
        "deviceId": "web-server",
        "deviceName": "Web Server (web-javascript)",
        "showMemoryUsage": false,
        "debugServer": 41239,
        "debuggerType": 2
    }
[11:26:58 AM] [General] [Warn] Did not find session ae59411e-6f97-489a-a12d-ce695e95e438 to handle dart.progressStart. There were 0 sessions:
[11:26:58 AM] [General] [Warn] Event will be queued and processed when the session start event fires
[11:26:58 AM] [General] [Info] Processing delayed event dart.progressStart for session ae59411e-6f97-489a-a12d-ce695e95e438
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] Spawning /home/egrimes/opt/flutter/bin/flutter with args ["run","--machine","--target","lib/main.dart","-d","web-server","--track-widget-creation","--dart-define=flutter.inspector.structuredErrors=true","--start-paused","--web-server-debug-protocol","ws","--web-allow-expose-url","-v"]
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] ..  in /home/egrimes/documents/development/sandbox/demo
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] ..  with {"toolEnv":{"FLUTTER_HOST":"VSCode","PUB_ENVIRONMENT":"vscode.dart-code"}}
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)]     PID: 8861
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [   +8 ms] executing: [/home/egrimes/opt/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [  +25 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] 8fe7655ed20ffd1395f68e30539a847a01a30351
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] executing: [/home/egrimes/opt/flutter/] git tag --contains HEAD
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [ +134 ms] Exit code 0 from: git tag --contains HEAD
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] 1.19.0-4.3.pre
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [   +7 ms] executing: [/home/egrimes/opt/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [   +3 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] origin/beta
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] executing: [/home/egrimes/opt/flutter/] git ls-remote --get-url origin
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [   +2 ms] Exit code 0 from: git ls-remote --get-url origin
[11:26:58 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] https://github.com/flutter/flutter.git
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [ +146 ms] Unable to locate an Android SDK.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [   +7 ms] executing: [/home/egrimes/opt/flutter/] git rev-parse --abbrev-ref HEAD
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [   +2 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] beta
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [  +29 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [ +117 ms] Unable to locate an Android SDK.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [  +62 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [ +154 ms] Generating /home/egrimes/documents/development/sandbox/demo/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [{"event":"daemon.connected","params":{"version":"0.5.3","pid":9095}}]
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [{"event":"app.start","params":{"appId":"e182c54e-004a-433a-960e-cb888a400377","deviceId":"web-server","directory":"/home/egrimes/documents/development/sandbox/demo","supportsRestart":true,"launchMode":"run"}}]
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [  +82 ms] Launching lib/main.dart on Web Server in debug mode...
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [{"method":"app.exposeUrl","params":{"url":"http://127.0.0.1:35911/$debug"},"id":"1"}]
[11:26:59 AM] [General] [Info] Exposing URL: http://127.0.0.1:35911/%24debug
[11:26:59 AM] [General] [Info] Mapping URL: http://127.0.0.1:35911/%24debug
[11:26:59 AM] [General] [Info] Mapped URL: http://127.0.0.1:35911/%24debug
[11:26:59 AM] [General] [Info] Final URI: http://127.0.0.1:35911/$debug
[11:26:59 AM] [General] [Info] Final URL: http://127.0.0.1:35911/$debug
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] ==> [{"id":"1","result":{"url":"http://127.0.0.1:35911/$debug"}}]
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [ +173 ms] Updating assets
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [{"event":"app.progress","params":{"appId":"e182c54e-004a-433a-960e-cb888a400377","id":"0","progressId":null,"message":"Syncing files to device Web Server..."}}]
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [ +107 ms] Generating /home/egrimes/documents/development/sandbox/demo/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [  +32 ms] <- reset
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [   +6 ms] /home/egrimes/opt/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev --old_gen_heap_size=2000 /home/egrimes/opt/flutter/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root /home/egrimes/opt/flutter/bin/cache/flutter_web_sdk/ --incremental --target=dartdevc --debugger-module-names -Ddart.developer.causal_async_stacks=true -Dflutter.inspector.structuredErrors=true --output-dill /tmp/flutter_tools.TJMSNG/flutter_tool.URZGSY/app.dill --libraries-spec file:///home/egrimes/opt/flutter/bin/cache/flutter_web_sdk/libraries.json --packages .packages -Ddart.vm.profile=false -Ddart.vm.product=false --bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions --enable-asserts --track-widget-creation --filesystem-root /tmp/flutter_tools.TJMSNG/flutter_tools.BDYERV --filesystem-scheme org-dartlang-app --initialize-from-dill build/cache.dill.track.dill --platform file:///home/egrimes/opt/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk.dill
[11:26:59 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [   +3 ms] <- compile org-dartlang-app:/web_entrypoint.dart
[11:27:13 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [{"event":"app.progress","params":{"appId":"e182c54e-004a-433a-960e-cb888a400377","id":"0","progressId":null,"finished":true}}]
[11:27:13 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [+13583 ms] Synced 22.6MB.
[11:27:13 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] <- accept
[11:27:13 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [        ] Waiting for connection from Dart debug extension at http://localhost:44955
[11:27:13 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [{"event":"app.webLaunchUrl","params":{"url":"http://localhost:44955","launched":false}}]
[11:27:13 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [{"method":"app.exposeUrl","params":{"url":"http://127.0.0.1:44956"},"id":"2"}]
[11:27:13 AM] [General] [Info] Exposing URL: http://127.0.0.1:44956/
[11:27:13 AM] [General] [Info] Mapping URL: http://127.0.0.1:44956/
[11:27:13 AM] [General] [Info] Mapped URL: http://127.0.0.1:54531/
[11:27:13 AM] [General] [Info] Final URI: http://127.0.0.1:54531/
[11:27:13 AM] [General] [Info] Final URL: http://127.0.0.1:54531/
[11:27:13 AM] [FlutterRun] [Info] [Flutter (Web Server)] ==> [{"id":"2","result":{"url":"http://127.0.0.1:54531/"}}]
[11:27:21 AM] [FlutterRun] [Info] [Flutter (Web Server)] <== [+7693 ms] Not connected to an app with id: 5K7kMZlBfTVBw9FzP0z1DQ==
Fri Jul 03 2020 [11:27:29 GMT-0500 (Central Daylight Time)] Log file ended

```

**Comment by DanTup** [comment]
@ErikGrimes I guess these are the port numbers you're referring to:

```
[{"event":"app.webLaunchUrl","params":{"url":"http://localhost:44955","launched":false}}]
[{"method":"app.exposeUrl","params":{"url":"http://127.0.0.1:44956"},"id":"2"}]
```

These ports are both actually valid - the first one is where Flutter is hosting the web server to serve up the web page assets, and the second one is where DWDS (the service that handles debugging) is hosting its backend server.

Both of them should be tunnelled by VS Code, and the URLs launched in the browser (and the injected URI for the debugger backend) should reflect the mapped URLs so they're accessible to the host (Windows) machine.

The URL in the `app.webLaunchUrl` event is the one that should've been launched in your browser, and should show a white/blank page until you click the Dart Debug extensions icon in Chrome.

If that's not what you're seeing, please let me know.

**Comment by ErikGrimes** [comment]
@DanTup 

It's launching the app.exposeUrl in the browser.  The app lauches as expected when I manually navigate to the the app.webLaunchUrl.

**Comment by DanTup** [comment]
@ErikGrimes 

> It's launching the app.exposeUrl in the browser.

Is this automatic, or are you clicking the link in the debug console? The browser should open automatically([the code is relatively basic](https://github.com/Dart-Code/Dart-Code/blob/1fe5a8185603f27a2065fbda575599669cd7c9d5/src/extension/commands/debug.ts#L464-L473)) but I'll see if I'm able to get the Windows update that includes WSL2 on my PC yet and try to repro using that (I tried with Remote Containers/Docker but it works as expected).

**Comment by DanTup** [comment]
I got WSL2 set up, and I was able to reproduce this on the Flutter beta branch - I had a similar error printed to the one above:

```
Not connected to an app with id: d6iKg/nrFg6oyr53b7wybg==
```

However, I was unable to repro on dev or master channels, so I think whatever was causing it may have been fixed.

I did see the ports being one-off from what was printed in the console, and what opened when I ctrl+clicked. I believe that's just a side-effect of how VS Code/Windows/WSL is mapping the port. Copy/pasting might not work, but Ctrl+Clicking will. This isn't unique to Dart/Flutter, it'll be the same when running other servers in WSL/GitPod/Codespaces/Containers, as you're seeing the raw output from the process in the remove machine.

Once https://github.com/flutter/flutter/issues/59463is fixed, there should be no need to manually open the window again.

**Comment by mymikemiller** [comment]
Just dropping by to say I'm still experiencing this exact issue on the current Beta channel. I'm using flutter for web, so I have no choice but to use Beta.

Just as in @ErikGrimes 's case, neither Web Server nor Chrome are working for me, with the same symptoms, though I'm also getting an error popup when clicking the Dart extension:

```
No Dart application detected. Your development server should inject metadata to indicate support for Dart debugging. This may require setting a flag. Check the documentation for your development server.
```

I haven't had to change any port numbers, though. If I relaunch enough times, it eventually works. I'm doing this on a 2012 mac mini while my laptop is in for repair. I suspect that maybe the problem might be my computer's speed. I see some logs like `background.js:426 [Violation] 'setTimeout' handler took 62981ms`

**Comment by DanTup** [comment]
@mymikemiller this issue was about WSL so I don't think it's the same issue as yours. I suspect you're right about the performance - if you click the Dart Debug extension too quickly it can fail with that message, though it sounds like that period is significantly worse for you.

If you're running locally, it's definitely a better experience to use the Chrome device and not web-server - it doesn't need the debug extension and should not have the same delay. Do you have problems using that? If so, is it only when running from VS Code or do you have the same issue if running from the terminal with `flutter run`?

**Comment by daniyalahmad358** [comment]
@DanTup  It works fine if I run it from the terminal, but doesn't worked when launched from the vs code run button.

**Comment by DanTup** [comment]
@daniyalahmad358 since there have been multiple issues mentioned here and this one is closed, please file a new issue and include detailed steps of what you're doing and what you're seeing. Running the **Dart: Capture Debugging Logs** command before starting to run the app and then clicking **Stop** on the logging notification after the problem has occurred will produce a helpful log file too (please ensure there's nothing sensitive in it before sharing).

Thanks!
